CMS/static/js/bookmark-nav.js

```diff
@@ -0,0 +1,40 @@
+(function ($) {
+    var NavBookMarkBtn = function(btn) {
+        this.btn = $(btn);
+        this.setup();
+    }
+
+    NavBookMarkBtn.prototype = {
+        setup: function() {
+            this.navBar = $('.discover-uni-nav');
+            this.loadSelectedCourses();
+            this.startWatcher();
+        },
+
+        loadSelectedCourses: function() {
+            if (JSON.parse(localStorage.getItem('comparisonCourses'))) {
+                this.selectedCourses = JSON.parse(localStorage.getItem('comparisonCourses'));
+            } else {
+                this.selectedCourses = [];
+            }
+            this.btn.text(this.selectedCourses.length);
+        },
+
+        startWatcher: function() {
+            var that = this;
+            this.navBar.on('loadeddata', function() {
+                that.loadSelectedCourses();
+            })
+        }
+    }
+
+    function init() {
+        var bookmarkBtns = $('.nav-bookmark__count');
+        for (var i = 0; i < bookmarkBtns.length; i++) {
+            new NavBookMarkBtn(bookmarkBtns[i]);
```

Comment:
Looks a little odd creating an object, but not storing any reference to it. I guess you're using some side-effect of the object creation.

Reply:
I'd never really thought about it, but I guess it is a bit odd.
Aside from the initial setup of the page the functionality of the objects is trigger by user interaction with the DOM rather than functions being programmatically called, which is why storing the reference is not needed. The interactions from the user are watched for by functions that are started during the initialisation of the object.

Reply:
Thanks Matt 👍